src/overlays/actors/ovl_Boss_Goma/z_boss_goma.c

```diff
@@ -21,115 +84,1717 @@ const ActorInit Boss_Goma_InitVars = {
+        Matrix_MultVec3f(&D_8091B3F4, &this->unk_260);
+    } else if (arg1 == 14) {
+        Matrix_MultVec3f(&D_8091B3F4, &this->unk_26C);
+    } else if (arg1 == 5) {
+        Matrix_MultVec3f(&D_8091B40C, &this->actor.posRot2.pos);
+    } else if (arg1 == 21) {
+        Matrix_MultVec3f(&D_8091B400, &this->unk_278);
+    } else if (arg1 == 73) {
+        Matrix_MultVec3f(&D_8091B400, &this->unk_284);
+    }
+
+    if (this->unk_1B8 == 2) {
+        if (*arg2 != NULL) {
+            Matrix_MultVec3f(&D_8091B400, &this->unk_2A8[arg1]);
+        } else {
+            this->unk_2A8[arg1].y = 10000.0f;
+        }
+    }
+
+    if (this->unk_758[arg1] == 1) {
+        this->unk_758[arg1] = 2;
+        Matrix_MultVec3f(&D_8091B418, &a);
+        Matrix_Get(&c);
+        func_800D20CC(&c, &b, MTXMODE_NEW);
+        tmp = (BossGoma*)Actor_SpawnAttached(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_GOMA, a.x, a.y,
+                                             a.z, b.x, b.y, b.z, D_8091B244[arg1] + 0x64);
+        if (tmp != NULL) {
+            tmp->unk_308 = *arg2;
+            tmp->actor.objBankIndex = this->actor.objBankIndex;
+        }
+    }
+    func_800628A4(arg1, &this->unk_7BC);
+}
+
+Gfx* func_8091ABEC(GraphicsContext* gfxCtx) {
+    Gfx* displayList;
+    Gfx* displayListHead;
+
+    displayList = displayListHead = Graph_Alloc(gfxCtx, 8);
+    gSPEndDisplayList(displayListHead++);
+
+    return displayList;
+}
+
+Gfx* func_8091AC18(GraphicsContext* gfxCtx) {
+    Gfx* displayListHead;
+    Gfx* displayList;
+
+    displayListHead = displayList = Graph_Alloc(gfxCtx, 4 * sizeof(Gfx));
+    gDPPipeSync(displayListHead++);
+    gDPSetRenderMode(displayListHead++, G_RM_PASS, G_RM_AA_ZB_TEX_EDGE2);
+    gSPClearGeometryMode(displayListHead++, G_CULL_BACK);
+    if (0) {};
+    gSPEndDisplayList(displayListHead++);
+    return displayList;
+}
+
+void BossGoma_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BossGoma* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_8091A9E8.s")
+    {
+        GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+        Gfx* dispRefs[4];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_8091ABEC.s")
+        Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_boss_goma.c", 4991);
+        func_80093D18(globalCtx->state.gfxCtx);
+        Matrix_Translate(0.0f, -4000.0f, 0.0f, MTXMODE_APPLY);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/func_8091AC18.s")
+        if (this->unk_1C2) {
+            gSPSegment(gfxCtx->polyOpa.p++, 0x08, func_8091AC18(globalCtx->state.gfxCtx));
+        } else {
+            gSPSegment(gfxCtx->polyOpa.p++, 0x08, func_8091ABEC(globalCtx->state.gfxCtx));
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Goma/BossGoma_Draw.s")
+        SkelAnime_Draw(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, func_8091A560, func_8091A9E8,
```

Comment:
These functions should be properly renamed the way we do for other actors that set up these callbacks.

Reply:
Re-opening since this wasn't addressed.